pil-analyzer/src/statement_processor.rs

```diff
@@ -426,110 +426,169 @@ where
             degree: self.degree,
         };
 
-        if let Some(FunctionDefinition::TypeDeclaration(enum_decl)) = value {
-            // For enums, we add PILItems both for the enum itself and also for all
-            // its type constructors.
-            assert_eq!(symbol_kind, SymbolKind::Other());
-            let enum_decl = self.process_enum_declaration(enum_decl);
-            let shared_enum_decl = Arc::new(enum_decl.clone());
-            let var_items = enum_decl.variants.iter().map(|variant| {
-                let var_symbol = Symbol {
-                    id: self.counters.dispense_symbol_id(SymbolKind::Other(), None),
-                    source: source.clone(),
-                    absolute_name: self
-                        .driver
-                        .resolve_namespaced_decl(&[&name, &variant.name])
-                        .relative_to(&Default::default())
-                        .to_string(),
-                    stage: None,
-                    kind: SymbolKind::Other(),
-                    length: None,
-                    degree: None,
-                };
-                let value = FunctionValueDefinition::TypeConstructor(
-                    shared_enum_decl.clone(),
-                    variant.clone(),
-                );
-                PILItem::Definition(var_symbol, Some(value))
-            });
-            return iter::once(PILItem::Definition(
-                symbol,
-                Some(FunctionValueDefinition::TypeDeclaration(enum_decl.clone())),
-            ))
-            .chain(var_items)
+        match value {
+            Some(value) => match value {
+                FunctionDefinition::TypeDeclaration(enum_decl) => {
+                    assert_eq!(symbol_kind, SymbolKind::Other());
+                    self.handle_enum_declaration_symbol(source, name, symbol, enum_decl)
+                }
+                FunctionDefinition::TraitDeclaration(trait_decl) => {
+                    self.handle_trait_declaration_symbol(source, name, symbol, trait_decl)
+                }
+                FunctionDefinition::Expression(expr) => {
+                    self.handle_expression_symbol(symbol_kind, symbol, type_scheme, expr)
+                }
+                FunctionDefinition::Array(value) => {
+                    self.handle_array_symbol(symbol, type_scheme, value)
+                }
+            },
+            None => vec![PILItem::Definition(symbol, None)],
+        }
+    }
+
+    fn handle_enum_declaration_symbol(
+        &mut self,
+        source: SourceRef,
+        name: String,
+        symbol: Symbol,
+        enum_decl: EnumDeclaration<parsed::Expression>,
+    ) -> Vec<PILItem> {
+        let enum_decl = self.process_enum_declaration(enum_decl);
+        let shared_enum_decl = Arc::new(enum_decl.clone());
+
+        let inner_items = enum_decl
```

Comment:
Maybe it would be better to move this code into the process_enum_declaration function (which is tiny) instead of creating a new function with a similar name?